src/include/sof/math/exp_fcn.h

```diff
@@ -13,8 +13,12 @@
 #if defined(__XCC__)
 /* HiFi */
 #include <xtensa/config/core-isa.h>
-#if XCHAL_HAVE_HIFI4 == 1
+#if XCHAL_HAVE_HIFI5 == 1
```

Comment:
Update also commit message of this patch, since this is adding the hifi3 version and some hifi4 & 5 logistics.

Reply:
Done

Reply:
Why is this ifdef needed? you can just check the HiFi directly in the file

Reply:
Then check is then required in all C code and HiFi intrinsic.

Depending on performance, build support ranges from 3 to 5 with a single common placed `#ifdef` that divides Generic C code and HiFi intrinsic that spans over all active settings. The test is being run by cmocka test using the standard function `sofm_exp_int32()`. will also need to bring along more #ifdef gear.

Consequently, I believe #ifdef are necessary.

Reply:
It's useful to have the added layer of generic/hifiN macros to e.g. build generic version with hifi toolchain. It could be simplified a bit but since it's only macros and actual code I'd not require perfection.

Reply:
@singalsu we have a bit of a mess around HIFI definitions atm e.g.
```
grep -rn HIFI4 src/audio/
src/audio/src/src_hifi4.c:11:#if SRC_HIFI4
src/audio/module_adapter/module/volume/volume_hifi4.c:24:#ifdef VOLUME_HIFI4
src/audio/module_adapter/module/volume/volume.c:1211:#ifdef VOLUME_HIFI4
src/audio/module_adapter/module/volume/volume.c:1246:#if XCHAL_HAVE_HIFI3 || XCHAL_HAVE_HIFI4
src/audio/module_adapter/module/volume/volume_hifi4_with_peakvol.c:9: * \brief Volume HIFI4 processing implementation with peak volume detection
src/audio/module_adapter/module/volume/volume_hifi4_with_peakvol.c:24:#ifdef VOLUME_HIFI4
src/audio/mixer/mixer_hifi3.c:10:#if __XCC__ && (XCHAL_HAVE_HIFI3 || XCHAL_HAVE_HIFI4)
src/audio/mixer/mixer.c:199:#if XCHAL_HAVE_HIFI3 || XCHAL_HAVE_HIFI4
src/audio/dcblock/dcblock_hifi4.c:12:#ifdef DCBLOCK_HIFI4
src/audio/copier/copier_hifi.c:8:#if __XCC__ && (XCHAL_HAVE_HIFI3 || XCHAL_HAVE_HIFI4)
src/audio/component.c:26:#if XCHAL_HAVE_HIFI3 || XCHAL_HAVE_HIFI4
```
This is not the PR to fix those, but we do need a high level SIMD level Kconfig that could be used by all modules and could also be used to downgrade if needed (for debug).

Reply:
> It's useful to have the added layer of generic/hifiN macros to e.g. build generic version with hifi toolchain.

Code speaks louder than words, here's an example from @singalsu (thanks!)
- #8108